### ISSUE TITLE
progress callback for clone

### ISSUE THREAD

**Issue by lowlander** [libgit2/pygit2#567]
Hello,

I am trying to get a progress callback working with the ```clone_repository``` call. 

I pass the following function 

```
	def _init_remote_cb(repo, name, url): 
		remote = repo.remotes.create(name, url)
		remote.transfer_progress = transfer_progress
		return remote
```

to the ```remote``` argument of ```clone_repository```, I see that this function is called, but there are never any calls to ```transfer_progress```.

What am I doing wrong?
             
PS: Is this even the right place for questions like this ?

**Comment by carlosmn** [comment]
What kind of url are you using? If you're using a local path, there won't be any packfile transfer. Instead git/libgit2 will copy over the object directory, bypassing the Git logic.

If you're using an older version of libgit2 (pre-0.23 IIRC) a non-local clone for a path on the filesystem won't generate these messages either.

**Comment by lowlander** [comment]
I am on 0.23.0 with libgit2 0.23.0. And I am using a remote git@1.2.3.4:/bla  gitolite/ssh URL. As a hack I tried to do a repository_init, add the remote and than do a fetch, and than the callbacks are called. 

BTW I noticed that before every fetch I have to reset my credentials;
```
rem.credentials = authentication_cb
tp = rem.fetch()
```

Is that normal/expected ?  Because during clone those credentials are already passed. 

Sadly I am not that fluent in python especially not when is comes to calling C from inside python. What would be the best way to debug this problem?

**Comment by carlosmn** [comment]
The custom remote initialization callback is there to set the configuration for the remote which `git_clone()` will use. Only the `git_remote` object is returned to libgit2. Anything you set on the python wrapper will not be propagated back. There is in fact nowhere to store it in the remote.

The functions you can override in the remote are kept in the python object so they will only be useful for `Remote.fetch()` and `Remote.push()`; clone can only make use of the options passed in via the function call itself.

This asymmetry is unfortunate, and it comes down to the old way in which libgit2 made you set the callbacks. I've been meaning to update how we handle this by using a dictionary or some form of object which can implement an interface but time is not plentiful.

**Comment by jdavid** [comment]
This issue has been fixed with PR #568 no ?

**Comment by lowlander** [comment]
It seems to work now with clone and fetch.
